ui/inspectors/inspector.reel/inspector-confirmation.reel/inspector-confirmation.js

```diff
@@ -8,9 +8,47 @@ var Component = require("montage/ui/component").Component;
  * @extends Component
  */
 exports.InspectorConfirmation = Component.specialize(/** @lends InspectorConfirmation# */ {
-    constructor: {
-        value: function InspectorConfirmation() {
-            this.super();
+
+    handleConfirmDeleteAction: {
+        value: function (event) {
+            this.parentComponent.isConfirmationVisible = false;
+            this.confirmDelete();
+        }
+    },
+
+    handleCancelDeleteAction: {
+        value: function() {
+            this.parentComponent.isConfirmationVisible = false;
+        }
+    },
+
+    confirmDelete: {
+        value: function (event) {
+            var self = this,
+                promise;
+
+            this.parentComponent._isToBeDeleted = true;
+
+            if (typeof this.parentComponent.parentComponent.delete === 'function') {
+                promise = this.parentComponent.parentComponent.delete();
+
+                if (Promise.is(promise)) {
+                    promise.catch(this.parentComponent._logError);
+                }
+            } else if (this.parentComponent.object) {
```

Comment:
@pchaussalet @Fzzr I see and it was already the case before.
So, I see two problems here: 
- The inspector probably need a sort of controller property, which could be by default the ownerComponent or might have been set within the serialization.
- The Overlay's Inspector needs a pointer to its "inspector/handler".

Keep in mind that later Montage will provide an application modal system in order to have just one modal overlay by application.